winter_messaging_transactional/producer/publish_processor.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import signal
+from threading import Event
+
+from injector import inject
+
+from winter_messaging_transactional.rabbitmq import TopologyConfigurator
+from winter_messaging_transactional.rabbitmq.rabbitmq_client import MessageNotPublishedException
+from winter_messaging_transactional.rabbitmq.rabbitmq_client import RabbitMQClient
+from winter_messaging_transactional.producer.outbox.outbox_message_dao import OutboxMessageDAO
+
+log = logging.getLogger(__name__)
+
+
+class PublishProcessor:
+    SLEEP_TIME = 1.0
+
+    @inject
+    def __init__(
+        self,
+        rabbitmq_client: RabbitMQClient,
+        outbox_message_doa: OutboxMessageDAO,
+        configurator: TopologyConfigurator,
+    ) -> None:
+        self._rabbitmq_client = rabbitmq_client
+        self._outbox_message_doa = outbox_message_doa
+        self._topology_configurator = configurator
+
+    def run(self):
+        log.info('Publishing processor started with sleep time: %s', self.SLEEP_TIME)
+        cancel_token = Event()
+
+        def handle_interrupt_signal(signum, frame):
+            cancel_token.set()
+
+        signal.signal(signal.SIGTERM, handle_interrupt_signal)
+        signal.signal(signal.SIGINT, handle_interrupt_signal)
+
+        is_error_occurred = False
+        while not cancel_token.wait(self.SLEEP_TIME):
+            outbox_messages = self._outbox_message_doa.select_unsent()
+            for index, outbox_message in enumerate(outbox_messages):
+                exchange = self._topology_configurator.get_exchange_key(outbox_message.topic)
```

Comment:
@mikhaillazko  Why do we use `topology_configurator.get_exchange_key` instead of `get_exchange_name from naming_convention` here? 

I mean, if publisher service doesn't have an event_handler for an event from a topic, then `topology_configurator` doesn't have a name for that exchange.

Reply:
exchange creates in advance for all topics that have been discovered and topics related to events. it means when you create new event and annotate it with topic then system should be configured on start and create exchange==topic. Exchange it's part of publisher side and it doesn't matter do we have consumer or not.